packages/uni_ui/lib/cards/restaurant_card.dart

```diff
@@ -0,0 +1,232 @@
+            child: IconButton(
+                onPressed: () {
+                  setState(() {
+                    isFavorite = !isFavorite;
+                  });
+                  if (widget.onFavoriteToggle != null) {
+                    widget.onFavoriteToggle!();
+                  }
+                },
+                icon: isFavorite
+                    ? UniIcon(PhosphorIconsFill.heart,
+                        color: Theme.of(context).primaryColor)
+                    : UniIcon(PhosphorIconsRegular.heart,
+                        color: Theme.of(context).primaryColor)),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class RestaurantCardMenuItem extends StatelessWidget {
+  const RestaurantCardMenuItem(
+      {super.key, required this.menuItemName, required this.menuItemType});
+
+  final String menuItemName;
+  final String menuItemType;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.symmetric(horizontal: 0, vertical: 8.0),
+      child: Row(
+        crossAxisAlignment: CrossAxisAlignment.center,
+        children: [
+          Expanded(
+              flex: 1,
+              child: MenuItemIcons.getMenuItemIcon(context, menuItemType)),
+          Expanded(
+            flex: 5,
+            child: Text(
+              menuItemName,
+              style: Theme.of(context).textTheme.bodyLarge,
+              softWrap: true,
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class RestaurantIcons {
+  static const cantina = PhosphorIconsDuotone.cookingPot;
+  static const grill = PhosphorIconsDuotone.cookingPot;
+  static const restaurant = PhosphorIconsDuotone.chefHat;
+  static const snackBar = PhosphorIconsDuotone.hamburger;
+
+  static UniIcon getRestaurantIcon(
+      BuildContext context, String restaurantType) {
+    var iconType;
+    switch (restaurantType) {
+      case "cantina":
+        iconType = cantina;
+        break;
+      case "grill":
+        iconType = grill;
+        break;
+      case "restaurant":
+        iconType = restaurant;
+        break;
+      case "snackBar":
+        iconType = snackBar;
+        break;
+      default:
+        iconType = restaurant;
+    }
+    return UniIcon(iconType, color: Theme.of(context).primaryColor);
+  }
+}
+
+class MenuItemIcons {
+  static const cow = PhosphorIconsDuotone.cow;
+  static const fish = PhosphorIconsDuotone.fish;
+  static const vegetarian = PhosphorIconsDuotone.plant;
+  static const soup = PhosphorIconsDuotone.bowlSteam;
+
+  static UniIcon getMenuItemIcon(BuildContext context, String menuItemType) {
+    var iconType;
+    switch (menuItemType) {
+      case "cow":
+        iconType = cow;
+        break;
+      case "fish":
+        iconType = fish;
+        break;
+      case "vegetarian":
+        iconType = vegetarian;
+        break;
+      case "soup":
+        iconType = soup;
+        break;
+      default:
+        iconType = cow;
+    }
```

Comment:
According to previous discussions, `uni_app` should "send" the icons to the RestaurantCard component. Therefore, the icons should be passed as part of the MenuItem object.